public/twilio-video.js

```diff
@@ -0,0 +1,173 @@
+class TwilioVideoPrototype extends HTMLElement {
+  // Fetch the access token from the server
+  fetchToken(identity) {
+    return fetch('/token?identity=' + identity).then(function(data){
+      return data.json();
+    });
+  }
+
+  // Initialise an access manager with the token, then initialise a conversation
+  // client using that access manager and listen for incoming invites.
+  connectToRoom(data) {
+    var roomName = this.getAttribute('room-name') || 'test';
+    console.log('Joining room "' + roomName + '"...');
+    var connectOptions = {
+      name: roomName,
+      logLevel: 'debug'
+    };
+    // Join the Room with the token from the server and the
+    // LocalParticipant's Tracks.
+    return Twilio.Video.connect(data.token, connectOptions);
+  }
+
+  // Prepare the behaviour for a conversation client when an invite is received.
+  addParticipants(room) {
+    this.activeRoom = room;
+
+    // Attach LocalParticipant's Tracks, if not already attached.
+    if (!this.me.querySelector('video')) {
+      this.attachParticipantTracks(room.localParticipant, this.me);
+    }
+
+    return room;
+  }
+
+  addRoomListeners(room) {
+    // Attach the Tracks of the Room's Participants.
+    room.participants.forEach(function(participant) {
+      console.log('Already in Room: "' + participant.identity + '"');
+      this.attachParticipantTracks(participant, this.caller);
+    }.bind(this));
+
+    // When a Participant joins the Room, log the event.
+    room.on('participantConnected', function(participant) {
+      console.log('Joining: "' + participant.identity + '"');
+    });
+
+    // When a Participant adds a Track, attach it to the DOM.
+    room.on('trackAdded', function(track, participant) {
+      console.log(participant.identity + ' added track: ' + track.kind);
+      this.attachTracks([track], this.caller);
+    }.bind(this));
```

Comment:
Could use fat arrow functions rather than binding, especially as we're within an ES2015 class here.